apps/roam/src/utils/getLeftSidebarSettings.ts

```diff
@@ -0,0 +1,175 @@
+import { RoamBasicNode } from "roamjs-components/types";
+import {
+  BooleanSetting,
+  getUidAndBooleanSetting,
+  getUidAndStringSetting,
+  IntSetting,
+  getUidAndIntSetting,
+  StringSetting,
+} from "./getExportSettings";
+import { getSubTree } from "roamjs-components/util";
+import getCurrentUserDisplayName from "roamjs-components/queries/getCurrentUserDisplayName";
+
+type LeftSidebarPersonalSectionSettings = {
+  uid: string;
+  truncateResult: IntSetting;
+  folded: BooleanSetting;
+  alias: StringSetting | null;
+};
+
+export type LeftSidebarPersonalSectionConfig = {
+  uid: string;
+  text: string;
+  sectionWithoutSettingsAndChildren: boolean;
+  settings?: LeftSidebarPersonalSectionSettings;
+  children?: RoamBasicNode[];
+  childrenUid?: string;
+};
+
+type LeftSidebarGlobalSectionConfig = {
+  uid: string;
+  collapsable: BooleanSetting;
+  folded: BooleanSetting;
+  children: RoamBasicNode[];
+  childrenUid: string;
+};
+
+export type LeftSidebarConfig = {
+  global: LeftSidebarGlobalSectionConfig;
+  personal: {
+    uid: string;
+    sections: LeftSidebarPersonalSectionConfig[];
+  };
+};
+
+const getLeftSidebarGlobalSectionConfig = (
+  leftSidebarChildren: RoamBasicNode[],
+): LeftSidebarGlobalSectionConfig => {
+  const globalSectionNode = getSubTree({
+    tree: leftSidebarChildren,
+    key: "Global-Section",
+  });
+  const globalChildren = globalSectionNode?.children || [];
+
+  const childrenNode = getSubTree({
+    tree: globalChildren,
+    key: "Children",
+  });
+
+  return {
+    uid: globalSectionNode?.uid || "",
+    collapsable: getUidAndBooleanSetting({
+      tree: globalChildren,
+      text: "Collapsable",
+    }),
+    folded: getUidAndBooleanSetting({ tree: globalChildren, text: "Folded" }),
```

Comment:
Even though there is currently only 1 "setting" we should put this under a "Settings" header to allow for additional settings in the future.  This would also match the Personal-Section